tests/histogram.spec.ts

```diff
@@ -0,0 +1,180 @@
+				boundaries: [1, 2, 3, 4, 5],
+				counts: [1, 0, 0, 3, 0, 4],
+				sum: 34.5,
+				expectedMin: 1
+			},
+			{
+				description: "only first bucket has values",
+				boundaries: [1, 2, 3, 4, 5],
+				counts: [3, 0, 0, 0, 0, 0],
+				sum: 0.75,
+				expectedMin: 0.25
+			},
+			{
+				description: "only one bucket",
+				boundaries: [],
+				counts: [4],
+				sum: 8.8,
+				expectedMin: 2.2
+			},
+			{
+				description: "only one bucket",
+				boundaries: [],
+				counts: [1],
+				sum: 1.2,
+				expectedMin: 1.2
+			},
+			{
+				description: "only last bucket has values",
+				boundaries: [1,2,3,4,5],
+				counts: [0,0,0,0,0,3],
+				sum: 15.6,
+				expectedMin: 5
+			}
+		];
+
+		for (const testCase of cases) {
+			test(testCase.description, () => {
+				const summary = estimateHistogram({
+					attributes: {},
+					endTime: [0, 0],
+					startTime: [0, 0],
+					value: {
+						sum: testCase.sum,
+						buckets: {
+							boundaries: testCase.boundaries,
+							counts: testCase.counts
+						},
+						count: testCase.counts.reduce((p, c) => p + c)
+					}
+				});
+
+				assert.ok(summary);
+				expect(summary.min).toBeCloseTo(testCase.expectedMin);
+			});
+		}
+	});
+
+	describe("max estimation", () => {
+		const cases = [
+			{
+				description: "values in second bucket",
+				boundaries: [1, 2, 3, 4, 5],
+				counts: [0, 1, 0, 3, 2, 0],
+				sum: 21.2,
+				expectedMax: 5
+			},
+			{
+				description: "Last bucket has value, use the last boundary as estimation instead of Inf.",
+				boundaries: [1, 2, 3, 4, 5],
+				counts: [1, 0, 0, 3, 0, 4],
+				sum: 34.5,
+				expectedMax: 5
+			},
+			{
+				description: "Only the last bucket has values, use the mean (10.1) Otherwise, the max would be estimated as 5, and max >= avg would be violated.",
+				boundaries: [1, 2, 3, 4, 5],
+				counts: [0, 0, 0, 0, 0, 2],
+				sum: 20.2,
+				expectedMax: 10.1
+			},
+			{
+				description: "Just one bucket from -Inf to Inf, calculate the mean as max value.",
+				boundaries: [],
+				counts: [4],
+				sum: 8.8,
+				expectedMax: 2.2
+			},
+			{
+				description: "Just one bucket from -Inf to Inf, calculate the mean as max value.",
+				boundaries: [],
+				counts: [1],
+				sum: 1.2,
+				expectedMax: 1.2
+			},
+			{
+				description: "Max is larger than the sum, use the estimated boundary.",
+				boundaries: [0,5],
+				counts: [0,2,0],
+				sum: 2.3,
+				expectedMax: 5
+			},
+			{
+				description: "Only the first bucket has a value, use the upper bound.",
+				boundaries: [1,2,3,4,5],
+				counts: [3,0,0,0,0,0],
+				sum: 1.5,
+				expectedMax: 1
+			},
+			{
+				description: "Sum is smaller than the largest bucket bound, mean is larger. Use mean. Otherwise, max would be estimated as -5 and max >= avg would be violated",
+				boundaries: [-10,-5],
+				counts: [0,0,3],
+				sum: -7.5,
+				expectedMax: -2.5
+			}
+			// TODO right now the estimation returns null and this is skipped.
+			// should we not skip it?
```

Comment:
If there is no way to create this data point with this SDK, then we can skip it.